comp/core/workloadmeta/collectors/internal/crio/containers.go

```diff
@@ -93,98 +82,43 @@ func getContainerStatus(ctx context.Context, client crio.Client, containerID str
-	// First, try to get resources from containerStatus
-	if containerStatus != nil && containerStatus.GetResources() != nil && containerStatus.GetResources().GetLinux() != nil {
-		var cpuLimit *float64
-		var memLimit *uint64
-		cpuPeriod := float64(containerStatus.GetResources().GetLinux().GetCpuPeriod())
-		cpuQuota := float64(containerStatus.GetResources().GetLinux().GetCpuQuota())
-		memLimitInBytes := uint64(containerStatus.GetResources().GetLinux().GetMemoryLimitInBytes())
-
-		if cpuPeriod != 0 && cpuQuota != 0 {
-			limit := cpuQuota / cpuPeriod
-			cpuLimit = &limit
-		}
-		if memLimitInBytes != 0 {
-			memLimit = &memLimitInBytes
-		}
-		return cpuLimit, memLimit
-	}
-
-	if info == nil || info["info"] == "" {
-		log.Warn("Info map is nil or does not contain resource information")
-		return nil, nil
-	}
-
-	// Fallback to parsing resources from info if status resources are nil
-	var parsedInfo struct {
-		RuntimeSpec struct {
-			Linux struct {
-				Resources struct {
-					CPU struct {
-						Quota  int64 `json:"quota"`
-						Period int64 `json:"period"`
-					} `json:"cpu"`
-					Memory struct {
-						LimitInBytes int64 `json:"memoryLimitInBytes"`
-					} `json:"memory"`
-				} `json:"resources"`
-			} `json:"linux"`
-		} `json:"runtimeSpec"`
-	}
-
-	if err := json.Unmarshal([]byte(info["info"]), &parsedInfo); err != nil {
-		log.Warnf("Failed to parse resources from container info: %v", err)
-		return nil, nil
-	}
-
-	cpuPeriod := float64(parsedInfo.RuntimeSpec.Linux.Resources.CPU.Period)
-	cpuQuota := float64(parsedInfo.RuntimeSpec.Linux.Resources.CPU.Quota)
-	memLimitInBytes := uint64(parsedInfo.RuntimeSpec.Linux.Resources.Memory.LimitInBytes)
-
-	var cpuLimit *float64
-	var memLimit *uint64
-	if cpuPeriod != 0 && cpuQuota != 0 {
-		limit := cpuQuota / cpuPeriod
-		cpuLimit = &limit
-	}
-	if memLimitInBytes != 0 {
-		memLimit = &memLimitInBytes
-	}
-	return cpuLimit, memLimit
-}
-
 // getContainerImage retrieves and converts a container image to workloadmeta format.
 func getContainerImage(ctx context.Context, client crio.Client, imageSpec *v1.ImageSpec) workloadmeta.ContainerImage {
 	if imageSpec == nil {
 		log.Warn("Image spec is nil, cannot fetch image")
 		return workloadmeta.ContainerImage{}
 	}
-	image, err := client.GetContainerImage(ctx, imageSpec)
-	if err != nil || image == nil {
+	imageResp, err := client.GetContainerImage(ctx, imageSpec, false)
+	if err != nil || imageResp == nil || imageResp.GetImage() == nil {
 		log.Warnf("Failed to fetch image: %v", err)
 		return workloadmeta.ContainerImage{}
 	}
-
+	image := imageResp.GetImage()
 	imgID := image.GetId()
 	imgName := ""
 	if len(image.GetRepoTags()) > 0 {
 		imgName = image.GetRepoTags()[0]
 	}
 	wmImg, err := workloadmeta.NewContainerImage(imgID, imgName)
 	if err != nil {
-		log.Warnf("Failed to create image: %v", err)
+		log.Debugf("Failed to create image: %v", err)
```

Comment:
when it is an error you can use `%w` instead of `%v` (see: https://go.dev/blog/go1.13-errors)

Reply:
Switched all the `%v` to `%w` when error wrapping with`fmt.Errorf()`. However, left logs as %v.